### PR TITLE
[#19] [Backend] As a user, I can see the survey details when taking a survey

### LINKED ISSUE #19: [Backend] As a user, I can see the survey details when taking a survey
## Why

App must call backend API to fetch survey details

## Acceptance Criteria

- Make post request to `https://survey-api.nimblehq.co/api/v1/surveys/{id}` and handle response
- if successful, return the list of survey questions
- if not successful, return the error message from API

## Resources

- [API](https://documenter.getpostman.com/view/11835486/UVsPNPfU#f8528306-27b9-42e6-bac9-a8a6f8ee77c6)

### PR DESCRIPTION
**Pull request by ryan-conway**
- Closes https://github.com/nimblehq/flutter-ic-lydia-ryan/issues/19

## What happened 👀

- Created `GetSurveyDetailUseCase` and related repository and api methods
- Created unit tests for `GetSurveyDetailUseCase`

## Insight 📝

- We can base the models off what was done for `GetSurveysUseCase` and related classes 🙏🏻 

## Proof Of Work 📹

- Unit tests pass

### CODE REVIEW

File: lib/api/repository/survey_repository.dart [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import 'package:injectable/injectable.dart';
 import 'package:lydiaryanfluttersurvey/api/exception/network_exceptions.dart';
 import 'package:lydiaryanfluttersurvey/api/service/survey_service.dart';
+import 'package:lydiaryanfluttersurvey/model/response/survey_detail_response.dart';
 import 'package:lydiaryanfluttersurvey/model/response/surveys_response.dart';
 
 abstract class SurveyRepository {
   Future<SurveysResponse> getSurveys();
+  Future<SurveyDetailResponse> getSurveyDetails(String surveyId);
```
Comment (manh-t) [inline lib/api/repository/survey_repository.dart]: should it be singular? Also, can you add a line break between functions for better readability? 😄 
```suggestion
  
  Future<SurveyDetailResponse> getSurveyDetail(String surveyId);
```

File: lib/api/service/survey_service.dart [hunk 2]
```diff
@@ -10,4 +11,9 @@ abstract class SurveyService {
 
   @GET('/api/v1/surveys')
   Future<SurveysResponse> getSurveys();
+
+  @GET('/surveys/{surveyId}')
+  Future<SurveyDetailResponse> getSurveyDetails(
```
Comment (manh-t) [inline lib/api/service/survey_service.dart]: same
```suggestion
  Future<SurveyDetailResponse> getSurveyDetail(
```

File: lib/usecases/get_survey_details_use_case.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import 'package:injectable/injectable.dart';
+import 'package:lydiaryanfluttersurvey/api/repository/survey_repository.dart';
+import 'package:lydiaryanfluttersurvey/model/response/survey_detail_response.dart';
+
+import 'base/base_use_case.dart';
+
+@Injectable()
+class GetSurveyDetailsUseCase extends UseCase<SurveyDetailResponse, String> {
```
Comment (manh-t) [inline lib/usecases/get_survey_details_use_case.dart]: Please update the test case name as well 🙏 
```suggestion
class GetSurveyDetailUseCase extends UseCase<SurveyDetailResponse, String> {
```

File: test/usecases/get_survey_details_use_case_test.dart [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:lydiaryanfluttersurvey/model/response/survey_detail_response.dart';
+import 'package:lydiaryanfluttersurvey/usecases/base/base_use_case.dart';
+import 'package:lydiaryanfluttersurvey/usecases/get_survey_details_use_case.dart';
+import 'package:mockito/mockito.dart';
+
+import '../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('GetSurveyDetailsUseCaseTest', () {
```
Comment (manh-t) [inline test/usecases/get_survey_details_use_case_test.dart]:
```suggestion
  group('GetSurveyDetailUseCaseTest', () {
```

File: test/usecases/get_survey_details_use_case_test.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import 'package:flutter_test/flutter_test.dart';
```
Comment (manh-t) [inline test/usecases/get_survey_details_use_case_test.dart]: and this file name too

**Review by manh-t — COMMENTED** [review]
@ryan-conway please update the PR's description 🙋

**Review by manh-t — APPROVED** [review]

**Review by markgravity — APPROVED** [review]

**Pull request merged**